bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVGsonDTO.java

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.magentatv.internal;
+
+import static org.openhab.binding.magentatv.internal.MagentaTVBindingConstants.*;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+import com.google.gson.InstanceCreator;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * The {@link MagentaTVGsonDTO} class implements The MR returns event information every time the program changes. This
+ * information is mapped to various Thing channels and also used to catch the power down event for MR400 (there is no
+ * way to query power status). This class provides the mapping between event JSON and Java class using Gson.
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+public class MagentaTVGsonDTO {
+    /*
+     * Program information event is send by the MR when a channel is changed.
+     *
+     * Sample data:
+     * {"type":"EVENT_EIT_CHANGE","instance_id":26,"channel_code":"54","channel_num":"11","mediaId":"1221",
+     * "program_info": [ {"start_time":"2018/10/14 10:21:59","event_id":"9581","duration":"00:26:47",
+     * "free_CA_mode":false,"running_status":4, "short_event": [{"event_name":"Mysticons","language_code":"DEU",
+     * "text_char":"Die Mysticons..." } ]},
+     * {"start_time":"2018/10/14 10:48:46","event_id":"12204","duration":"00:23:54","free_CA_mode":false,
+     * "running_status":1, "short_event": [ {"event_name":"Winx Club","language_code":"DEU", "text_char":"Daphnes Eltern
+     * veranstalten...!" }]} ] }
+     */
+    // The following classes are used to map the JSON data into objects using GSon.
+    public static class MRProgramInfoEvent {
+        public String type = "";
+        @SerializedName("instance_id")
+        public Integer instanceId = 0;
+        @SerializedName("channel_code")
+        public String channelCode = "";
+        @SerializedName("channel_num")
+        public String channelNum = "";
+        public String mediaId = "";
```

Comment:
missing `@SerializedName`?

Reply:
nope, they are not always consistent :-)